File: server/src/Routes/RoleRoutes.js
import express from "express";
import JWT from "../Middlewares/jsonwebtoken.js";
import RoleController from "../Controllers/roleControlller.js";

const RoleRoutes = express.Router();

RoleRoutes.post("/create", JWT.validateAccessToken, RoleController.createRole);
RoleRoutes.get(
  "/get-all-field-roles",
  JWT.validateAccessToken,
  RoleController.getAllFieldRoles
);
RoleRoutes.get("/get-all", JWT.validateAccessToken, RoleController.getAllRoles);
RoleRoutes.get(
  "/get-one/:roleId",
  JWT.validateAccessToken,
  RoleController.getRole
);
RoleRoutes.put(
  "/update/:roleId",
  JWT.validateAccessToken,
  RoleController.updateRole
);
RoleRoutes.delete(
  "/delete/:roleId",
  JWT.validateAccessToken,
  RoleController.deleteRole
);

export default RoleRoutes;

File: server/src/routes/roleRoutes.go
package routes

import (
	"controll/src/controllers"

	"github.com/gin-gonic/gin"
)

func RoleRoutes(router *gin.RouterGroup) {
	role := router.Group("/api/role")
	{
		role.POST("/create", controllers.CreateRole())
		role.GET("/get-all-field-roles", controllers.GetAllFieldRoles())
		role.GET("/get-all", controllers.GetAllRoles())
		role.GET("/get-one/:roleId", controllers.GetRole())
		role.PUT("/update/:roleId", controllers.UpdateRole())
	}
}
